cylc/flow/taskdef.py

```diff
@@ -73,26 +73,44 @@ def generate_graph_children(tdef, point):
     return graph_children
 
 
-def generate_graph_parents(tdef, point):
-    """Determine graph parents of this task."""
+def generate_graph_parents(tdef, point, taskdefs):
+    """Determine concrent graph parents of task tdef at point.
+
+    Infer parents be reversing upstream triggers that lead to point/task.
+    """
     graph_parents = {}
-    for seq, ups in tdef.graph_parents.items():
+    for seq, triggers in tdef.graph_parents.items():
+        if not seq.is_valid(point):
+            # Don't infer parents if the trigger belongs to a sequence that
+            # does not include the child point. E.g.:
+            #   T06 = "waz[-PT6H] => foo"
+            # here waz[-PT6H] is a parent of T06/foo but not of T12/foo.
```

Comment:
This example is not making a lot of sense to me

Reply:
It's essentially the same as the example in the issue:

waz doesn't exist on the sequence T06m, it is a parent (i.e. upstream) of foo at T06